rotkehlchen/chain/substrate/manager.py

```diff
@@ -132,108 +119,111 @@ def __init__(
-        except RemoteError as e:
-            log.error(str(e))
-            raise
+        last_block = self._get_last_block(node_interface=node_interface)
 
         # Last block via Subscan API
-        chain_metadata = None
         try:
             chain_metadata = self._request_chain_metadata()
-        except RemoteError as e:
-            log.error(str(e))
+        except RemoteError:
             self.msg_aggregator.add_warning(
                 f'Unable to verify that {self.chain} node at endpoint {node_interface.url} '
                 f'is synced with the chain. Balances and other queries may be incorrect.',
             )
-        else:
-            # Check node synchronization
-            try:
-                metadata_last_block = BlockNumber(chain_metadata['data']['blockNum'])
-            except KeyError as e:
-                message = (
-                    f'{self.chain} failed to deserialize the chain metadata response: {str(e)}.'
-                )
-                log.error(message, chain_metadata=chain_metadata)
-                raise RemoteError(message) from e
+            return last_block
 
-            log.debug(
-                f'{self.chain} subscan API metadata last block',
-                metadata_last_block=metadata_last_block,
+        # Check node synchronization
+        try:
+            metadata_last_block = BlockNumber(chain_metadata['data']['blockNum'])
+        except KeyError as e:
+            message = f'{self.chain} failed to deserialize the chain metadata response: {str(e)}.'
+            log.error(message, chain_metadata=chain_metadata)
+            raise RemoteError(message) from e
+
+        log.debug(
+            f'{self.chain} subscan API metadata last block',
+            metadata_last_block=metadata_last_block,
+        )
+        if metadata_last_block - last_block > self.chain.blocks_threshold():
+            self.msg_aggregator.add_warning(
+                f'Found that {self.chain} node at endpoint {node_interface.url} '
+                f'is not synced with the chain. Node last block is {last_block}, '
+                f'expected last block is {metadata_last_block}. '
+                f'Balances and other queries may be incorrect.',
             )
-            if metadata_last_block - last_block > self.chain.get_blocks_threshold():
-                self.msg_aggregator.add_warning(
-                    f'Found that {self.chain} node at endpoint {node_interface.url} '
-                    f'is not synced with the chain. Node last block is {last_block}, '
-                    f'expected last block is {metadata_last_block}. '
-                    f'Balances and other queries may be incorrect.',
-                )
 
         return last_block
 
-    def _connect(self) -> None:
-        node_name = self.chain.get_node_name()
-        for node in node_name:
+    def _attempt_connections(self, node_names: Sequence[NodeName]) -> None:
+        for node in node_names:
```

Comment:
Something dawned on me as I was looking at this code.

Most of our users will not have a kusama address and as such will not care about this. Initializing an empty KusamaInterface object by doing almost nothing at the constructor is fine.

But initializing a bunch of connections if the user will never need them is not. So what would be useful here, would be to do this if the user has any kusama addresses.

And also do these connections only when the first kusama address is added. Does this make sense? You can definitely do this in another PR after this. Should be easy.

Same thing applies to ethereum by the way. We only did not think about it since all of our users use ethereum.

Reply:
Issue is here: https://github.com/rotki/rotki/issues/2080

Resolving the comment.